### ISSUE TITLE
Add the OSMC user into the 'input' group by default

### ISSUE THREAD

**Issue by kurac338** [osmc/osmc#748]
Hello, can we please add the `osmc` user into the `input` group by default in the OS?

In particular, it would stop any new users from having issues regarding [this topic](https://github.com/moonlight-stream/moonlight-qt/issues/1094) and as far as I can tell, should not impact anything negatively.

TL;DR: `osmc` user is not able to access the `hidraw` device (external BT controller) in order to utilize all of its features, due to not being in the `input` user group by default. Tested on 2023-08-01.

**Comment by samnazarko** [comment]
It may make more sense to make hidraw accessible by the osmc group otherwise this change would only take effect on new installs

Sam

On 28 Oct 2023, at 09:12, Mirko Brand ***@***.***> wrote:

﻿

Hello, can we please add the osmc user into the input group by default in the OS?

In particular, it would stop any new users from having issues regarding this topic<https://github.com/moonlight-stream/moonlight-qt/issues/1094> and as far as I can tell, should not impact anything negatively.

TL;DR: osmc user is not able to access the hidraw device (external BT controller) in order to utilize all of its features, due to not being in the input user group by default. Tested on 2023-08-01.

**Comment by cgutman** [comment]
FYI, I just picked the `input` group for these gamepad hidraw devices because it seemed like a reasonable group to pick on stock Debian. I don't think there's any real standard about what group should own these devices (probably depends on the nature of each HID device). I'd prefer to stick as close to stock Debian as possible though, since these are generic Debian packages. These are my udev rules: https://github.com/cgutman/moonlight-packaging/blob/master/debian/moonlight-qt.udev

I was trying to avoid enabling access to all hidraw devices to everyone, since that could have negative impacts on security for multi-user systems. However, for a single user OS like OSMC, maybe it makes sense to avoid having to hardcode VID/PID values and just set all hidraw devices to be owned by `osmc`/`input` or `0666`.

> It may make more sense to make hidraw accessible by the osmc group otherwise this change would only take effect on new installs

Maybe you could put a `useradd -a -G input osmc` in a `postinst` script for some core package to take care of upgrades. Not sure if that's feasible.

It looks like OSMC uses the `osmc` group for ownership of input devices (rather than `input`), so the rules in the package are going to conflict with OSMC's rules. I assume the rule priority is the right way to deal with that problem and I've simply used the default (60). As long as your rules supersede mine (or the OSMC user is in the `input` group), it should all be okay.
